pygmt/base_plotting.py

```diff
@@ -644,13 +644,18 @@ def legend(self, spec=None, **kwargs):
         {R}
         position (D) : str
             ``'[g|j|J|n|x]refpoint+wwidth[/height][+jjustify][+lspacing][+odx[/dy]]'``
-            Defines the reference point on the map for the legend.
+            Defines the reference point on the map for the legend. By default, uses
+            'JTR+jTR' which places the legend in the upper-right corner, inside.
         box (F) : bool or str
             ``'[+cclearances][+gfill][+i[[gap/]pen]][+p[pen]][+r[radius]][+s[[dx/dy/][shade]]]'``
             Without further options, draws a rectangular border around the
             legend using **MAP_FRAME_PEN**.
         """
         kwargs = self._preprocess(**kwargs)
+
+        if "D" not in kwargs:
+            kwargs["D"] = position
```

Comment:
This actually works amazingly enough! So let me get this straight:

- if the short alias "D" is set, then that is the one used.
- If the long alias 'position' is set, then it turns into "D" which is the one used.
- If the short alias "D" is not set, then the default position is used.
- If the long alias 'position' is not set, then there is no "D", so the default position is used.

However, if the short alias "D" and "position" are simultaneously set, there will be presumably be an error (good), or one overrides the other without erroring (bad)?

Reply:
Yes, all the above are correct. No error is given if both `D` and `position` are set — but that's not caused by this PR; the same silent overriding action occurs on the master branch too.

Python doesn't let folks repeat kwargs, but in this case we'd need to devise a way to check for repeats once long names are converted to short ones.

Reply:
Right, that'll be a bigger task for another time then.